cpp/src/io/parquet/reader_impl.cu

```diff
@@ -1222,6 +1237,17 @@ rmm::device_buffer reader::impl::decompress_page_data(
 
   decompress_check(comp_res, _stream);
 
+  // now copy the uncompressed V2 def and rep level data
+  if (not copy_in.empty()) {
+    host_span<device_span<uint8_t const> const> copy_in_view{copy_in.data(), copy_in.size()};
+    auto const d_copy_in = cudf::detail::make_device_uvector_async(copy_in_view, _stream);
+
+    host_span<device_span<uint8_t> const> copy_out_view(copy_out.data(), copy_out.size());
+    auto const d_copy_out = cudf::detail::make_device_uvector_async(copy_out_view, _stream);
+
+    gpu_copy_uncompressed_blocks(d_copy_in, d_copy_out, _stream);
```

Comment:
I think we need to sync before returning from `decompress_page_data` (only when `not copy_in.empty()`), because `copy_in` and `copy_out` may go out of scope before the copies are performed in `make_device_uvector_async` calls.

Reply:
Ok, I'll add a stream sync.